1.0/docs/devguide/tests.md

```diff
@@ -0,0 +1,367 @@
+    suite('<iron-ajax>', function() {
+      var ajax;
+      var request;
+      var server;
+      var responseHeaders = {
+        json: { 'Content-Type': 'application/json' }
+      };
+      setup(function() {
+        server = sinon.fakeServer.create();
+        server.respondWith(
+          'GET',
+          /\/responds_to_get_with_json.*/, [
+            200,
+            responseHeaders.json,
+            '{"success":true}'
+          ]
+        );
+      });
+      teardown(function() {
+        server.restore();
+      });
+      suite('when making simple GET requests for JSON', function() {
+        setup(function() {
+          // get fresh instance of iron-ajax before every test
+          ajax = fixture('simple-get');
+        });
+        test('has sane defaults that love you', function() {
+          request = ajax.generateRequest();
+          server.respond();
+          expect(request.response).to.be.ok;
+          expect(request.response).to.be.an('object');
+          expect(request.response.success).to.be.equal(true);
+        });
+        test('has the correct xhr method', function() {
+          request = ajax.generateRequest();
+          expect(request.xhr.method).to.be.equal('GET');
+        });
+      });
+    });
+  </script>
+</body>
+</html>
+{% endhighlight %}
+
+## Control which tests are run
+
+Use a test index to run a subset of suites.
+
+{% highlight html %}
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <script src=”../bower_components/webcomponentsjs/web-components-lite.js”></script>
+    <script src=”../bower_components/web-component-tester/browser.js”></script>
+  </head>
+  <body>
+    <script>
+      WCT.loadSuites([
+        'basic.html',
+        'async.html'
+      ]);
+    </script>
+  </body>
+</html>
+{% endhighlight %}
+
+You can also use test indexes to configure your tests via query strings
+when `wct` loads them. See [Test Shadow DOM](#shadow-dom) for an example.
+
+## Prevent shared state with test fixtures {#test-fixtures}
+
+Test fixtures enable you to define a template of content and copy a clean,
+new instance of that content into each test suite. Use test fixtures to
+minimize the amount of shared state between test suites. 
+
+To use a test fixture:
+
+*   Define the test fixture template and give it an ID.
+*   Define a variable in your test script to reference the template.
+*   Instantiate a new instance of the fixture in your `setup()` method.
+
+{% highlight html %}
+<test-fixture id="seed-element-fixture">
+  <template>
+    <seed-element>
+      <h2>seed-element</h2>
+    </seed-element>
+  </template>
+</test-fixture>
+
+<script>
+  suite('<seed-element>', function() {
+    var myEl;
+    setup(function() {
+      myEl = fixture('seed-element-fixture');
+    });
+    test('defines the "author" property', function() {
+      assert.equal(myEl.author.name, 'Dimitri Glazkov');
+    });
+  });
+</script>
+{% endhighlight %}
+
+## Make tests asynchronous
```

Comment:
This entry confuses me because the test is asynchronous, but it's testing something that isn't asynchronous. The question being, is this:

Asynchronous tests?
Test asynchronous actions?

I'm not really sure from the example how I'm supposed to use this.